core/common/src/main/java/alluxio/uri/Authority.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.uri;
+
+import java.io.Serializable;
+
+/**
+ * This interface represents the authority part of a URI.
+ */
+public interface Authority extends Comparable<Authority>, Serializable {
+  /**
+   * Gets the Authority object from the input string.
+   *
+   * @param authority the string authority to transfer
+   * @return an Authority object
+   */
+  static Authority fromString(String authority) {
+    if (authority == null || authority.length() == 0) {
+      return null;
+    }
+    String zkAuthorityPattern = "^zk@(.*)";
```

Comment:
- make this a static constant and  make it a precompiled pattern for efficiency (see https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html)
- actually is it redundant from `ZookeeperAuthority .ZOOKEEPER_PATTERN`? We should not define two identical constants

Reply:
Now only define it in the Authority.fromString and make it precompiled pattern.